modules/src/clients/ics07_tendermint/client_state.rs

```diff
@@ -454,6 +479,164 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn client_state_verify_delay_passed() {
+        use std::time::Duration;
+        #[derive(Debug, Clone)]
+        struct Params {
+            current_time: Timestamp,
+            current_height: Height,
+            processed_time: Timestamp,
+            processed_height: Height,
+            delay_period_time: Duration,
+            delay_period_blocks: u64,
+        }
+        struct Test {
+            name: String,
+            params: Params,
+            want_pass: bool,
+        }
+        let now = Timestamp::now();
+
+        let tests: Vec<Test> = vec![
+            Test {
+                name: "Successful delay verification".to_string(),
+                params: Params {
+                    current_time: (now + Duration::from_nanos(2000)).unwrap(),
+                    current_height: Height::new(0, 5),
+                    processed_time: (now + Duration::from_nanos(1000)).unwrap(),
+                    processed_height: Height::new(0, 3),
+                    delay_period_time: Duration::from_nanos(500),
+                    delay_period_blocks: 2,
+                },
+                want_pass: true,
+            },
+            Test {
+                name: "Delay period(time) has not elapsed".to_string(),
+                params: Params {
+                    current_time: (now + Duration::from_nanos(1200)).unwrap(),
+                    current_height: Height::new(0, 5),
+                    processed_time: (now + Duration::from_nanos(1000)).unwrap(),
+                    processed_height: Height::new(0, 3),
+                    delay_period_time: Duration::from_nanos(500),
+                    delay_period_blocks: 2,
+                },
+                want_pass: false,
+            },
+            Test {
+                name: "Delay period(blocks) has not elapsed".to_string(),
+                params: Params {
+                    current_time: (now + Duration::from_nanos(2000)).unwrap(),
+                    current_height: Height::new(0, 5),
+                    processed_time: (now + Duration::from_nanos(1000)).unwrap(),
+                    processed_height: Height::new(0, 4),
+                    delay_period_time: Duration::from_nanos(500),
+                    delay_period_blocks: 2,
+                },
+                want_pass: false,
+            },
+        ];
```

Comment:
I was playing with these tests and wanted to provoke an error to see how it looks. I used this new test, which should not pass (but I set it to `want_pass: false,` so that errors get displayed).

```rust
            Test {
                name: "Error".to_string(),
                params: Params {
                    current_time: now,
                    current_height: Height::new(0, 5),
                    processed_time: (now + Duration::from_secs(10)).unwrap(),
                    processed_height: Height::new(0, 4),
                    delay_period_time: Duration::from_secs(50),
                    delay_period_blocks: 2,
                },
                want_pass: true,
            },
```

When running the test I found was that our error definition is incomplete. Specifically, we output this error:

> ClientState::verify_delay_passed() failed for test Error, 
msgParams ... with error Some(not enough time elapsed, current timestamp {0} is still less than earliest acceptable timestamp {1}

Note the `{0}` and `{1}` which should be an interpolation but it is not. All the user sees is this raw information, which is not very useful.

I propose the diff below on how we can change the error type for `NotEnoughTimeElapsed` to provide correct information.

```diff
diff --git a/modules/src/clients/ics07_tendermint/error.rs b/modules/src/clients/ics07_tendermint/error.rs
index 54966531..ad46795d 100644
--- a/modules/src/clients/ics07_tendermint/error.rs
+++ b/modules/src/clients/ics07_tendermint/error.rs
@@ -160,7 +160,7 @@ define_error! {
                 current_time: Timestamp,
                 earliest_time: Timestamp,
             }
-            |_| { "not enough time elapsed, current timestamp {0} is still less than earliest acceptable timestamp {1}" },
+            | e | { format_args!("not enough time elapsed, current timestamp {0} is still less than earliest acceptable timestamp {1}", e.current_time, e.earliest_time) },
```

With this change, the error should look like this
> ... error Some(not enough time elapsed, current timestamp Timestamp(2022-02-15T08:49:13.695546Z) is still less than earliest acceptable timestamp Timestamp(2022-02-15T08:50:13.695546Z)

What I would suggest:

- @Wizdave97 can you please try to recreate my test above and change the definition of error type `NotEnoughTimeElapsed` in this PR so that it provides complete error details? Note that the error definition for `NotEnoughBlocksElapsed` is also incomplete and needs to be fixed.

- @soareschen please let us know if we're using `format_args` in flex-error correctly here.